### PR TITLE
Add reference to "Compiled Mode" in Variables/Expressions info page.

### PR DESCRIPTION
**Pull request by SudarshanPurohit**
A preview is available:

[Debugging expressions](https://64d5060049f47f02bc8099d8--harness-developer.netlify.app/docs/platform/Variables-and-Expressions/harness-variables#debugging-expressions)

This commit will also need a new screenshot file to be added as an image. Attaching it to PR. 

expression_evaluator_screen

# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*
expression_evaluator_screen

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 1]
```diff
@@ -444,6 +444,23 @@ For example, instead of `<+pipeline.variable.var1>_suffix`, use these syntaxes:
 - `<+<+pipeline.variables.var1>.concat("_suffix")>`
 - `<+<+pipeline.variables.var1> + "_suffix">` 
 
+## Debugging Expressions
+
+:::note 
+
+This feature is currently behind the Feature Flag PIE_EXPRESSION_PLAYGROUND. Please contact your Harness CSM to enable it for your account.
```
Comment (brian-f-harness) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:451]:
```suggestion
This feature is currently behind the feature flag `PIE_EXPRESSION_PLAYGROUND`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
```

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 2]
```diff
@@ -444,6 +444,23 @@ For example, instead of `<+pipeline.variable.var1>_suffix`, use these syntaxes:
...
+
+:::
+
+An easy way to debug expressions in your pipeline is to use the "Compiled Mode" in your Variable Panel. This is enabled using a radio button at the top of the Variable Panel. When switched on, all of the expressions in the panel are compiled and the values displayed. The compilation happens by default against the Pipeline's latest execution. You can change this by selecting from a displayed list of previous executions.
```
Comment (brian-f-harness) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:455]:
```suggestion
An easy way to debug expressions in your pipeline is to use Compiled Mode in your Variable Panel. You can enable this mode using a radio button at the top of the Variable Panel. When turned on, all of the expressions in the panel are compiled and the values are displayed. The compilation happens by default against the Pipeline's latest execution. You can change this by selecting from a displayed list of previous executions. 
```

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 3]
```diff
@@ -444,6 +444,23 @@ For example, instead of `<+pipeline.variable.var1>_suffix`, use these syntaxes:
...
+
+This feature is currently behind the Feature Flag PIE_EXPRESSION_PLAYGROUND. Please contact your Harness CSM to enable it for your account.
+
+:::
+
+An easy way to debug expressions in your pipeline is to use the "Compiled Mode" in your Variable Panel. This is enabled using a radio button at the top of the Variable Panel. When switched on, all of the expressions in the panel are compiled and the values displayed. The compilation happens by default against the Pipeline's latest execution. You can change this by selecting from a displayed list of previous executions. 
+
+![](./static/expression_evaluator_screen.png)
+
+Expressions that are incorrect or which cannot be evaluated using the Execution data are highlighted in the Variable values. You can then switch the panel back to normal mode and correct the expression.
```
Comment (brian-f-harness) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:459]:
```suggestion
Expressions that are incorrect or cannot be evaluated using the Execution data are highlighted in the **Variable** values. You can switch the panel back to normal mode and correct the expression. 
```

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 4]
```diff
@@ -444,6 +444,23 @@ For example, instead of `<+pipeline.variable.var1>_suffix`, use these syntaxes:
...
+
+:::
+
+An easy way to debug expressions in your pipeline is to use the "Compiled Mode" in your Variable Panel. This is enabled using a radio button at the top of the Variable Panel. When switched on, all of the expressions in the panel are compiled and the values displayed. The compilation happens by default against the Pipeline's latest execution. You can change this by selecting from a displayed list of previous executions. 
+
+![](./static/expression_evaluator_screen.png)
+
+Expressions that are incorrect or which cannot be evaluated using the Execution data are highlighted in the Variable values. You can then switch the panel back to normal mode and correct the expression. 
+
+In case you need to test out an expression that isn't a part of a variable (say, something in a script, you can create a temporary variable in the panel, assign the expression to it, and use the Compiled Mode to debug it.
```
Comment (brian-f-harness) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:461]:
```suggestion
To test out an expression that isn't part of a variable (say, something in a script), you can create a temporary variable in the panel, assign the expression to it, and use Compiled Mode to debug it. 
```

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 5]
```diff
@@ -444,6 +444,23 @@ For example, instead of `<+pipeline.variable.var1>_suffix`, use these syntaxes:
...
+
+:::note 
+
+This feature is currently behind the Feature Flag PIE_EXPRESSION_PLAYGROUND. Please contact your Harness CSM to enable it for your account.
+
+:::
+
+An easy way to debug expressions in your pipeline is to use the "Compiled Mode" in your Variable Panel. This is enabled using a radio button at the top of the Variable Panel. When switched on, all of the expressions in the panel are compiled and the values displayed. The compilation happens by default against the Pipeline's latest execution. You can change this by selecting from a displayed list of previous executions. 
+
+![](./static/expression_evaluator_screen.png)
```
Comment (brian-f-harness) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:457]:
```suggestion
![](./static/expression-evaluator-screen.png)
```

**Review by brian-f-harness — DISMISSED** [review]

**Review by brian-f-harness — APPROVED** [review]

**Review by deepakbolangady — APPROVED** [review]
Approving with some minor suggestions and feedback.

**Pull request merged**